test/integration/sapi-nv-readwrite.int.c

```diff
@@ -0,0 +1,185 @@
+
+    rc = Tss2_Sys_PolicyGetDigest (sys_ctx, auth_handle, 0, &policy_hash, 0);
+    return_if_error (rc, "Tss2_Sys_PolicyGetDigest");
+    LOGBLOB_INFO (policy_hash.buffer, policy_hash.size, "policy_hash");
+
+    rc = Tss2_Sys_NV_DefineSpace (sys_ctx,
+                                  TPM2_RH_PLATFORM,
+                                  &auth_cmd_null_pwd,
+                                  &nv_auth,
+                                  &public_info,
+                                  &auth_rsp);
+    return_if_error (rc, "Tss2_Sys_NV_DefineSpace");
+
+    rc = Tss2_Sys_FlushContext (sys_ctx, auth_handle);
+    return_if_error (rc, "Tss2_Sys_FlushContext");
+
+    return TSS2_RC_SUCCESS;
+}
+
+static TSS2_RC
+nv_write_read_test (TSS2_SYS_CONTEXT *sys_ctx,
+                    TPMI_RH_NV_INDEX index)
+{
+    TSS2_RC rc;
+    TPM2B_MAX_NV_BUFFER write_data = {
+        .size = 4,
+        .buffer = { 0xde, 0xad, 0xbe, 0xef },
+    };
+    TPM2B_MAX_NV_BUFFER nv_buf = { 0, };
+    TSS2L_SYS_AUTH_RESPONSE auth_resp = { 0, };
+
+    rc = TSS2_RETRY_EXP (Tss2_Sys_NV_Write (sys_ctx,
+                                            index,
+                                            index,
+                                            &auth_cmd_null_pwd,
+                                            &write_data,
+                                            0,
+                                            &auth_resp));
+    return_if_error (rc, "Tss2_Sys_NV_Write");
+
+    rc = Tss2_Sys_NV_Read (sys_ctx,
+                           index,
+                           index,
+                           &auth_cmd_null_pwd,
+                           4,
+                           0,
+                           &nv_buf,
+                           &auth_resp);
+    return_if_error (rc, "Tss2_Sys_NV_Read");
+
+    if (memcmp (nv_buf.buffer, write_data.buffer, write_data.size) != 0) {
+        LOG_ERROR ("%s: data read from NV is different from data written",
+                   __func__);
+        LOGBLOB_DEBUG (write_data.buffer, write_data.size, "write_data");
+        LOGBLOB_DEBUG (nv_buf.buffer, nv_buf.size, "nv_buf");
+        return 1;
+    }
+
+    return TSS2_RC_SUCCESS;
+}
+
+static TSS2_RC
+teardown_nv (TSS2_SYS_CONTEXT *sys_ctx,
+             TPMI_RH_NV_INDEX index)
+{
+    TSS2_RC rc;
+    TSS2L_SYS_AUTH_RESPONSE auth_resp = { 0, };
+
+    rc = Tss2_Sys_NV_UndefineSpace (sys_ctx,
+                                    TPM2_RH_PLATFORM,
+                                    index,
+                                    &auth_cmd_null_pwd,
+                                    &auth_resp);
+    return_if_error (rc, "Tss2_Sys_NV_UndefineSpace");
+
+    return TSS2_RC_SUCCESS;
+}
+int
+test_invoke (TSS2_SYS_CONTEXT *sys_ctx)
+{
+    TSS2_RC rc;
+
+    rc = setup_nv (sys_ctx, INDEX_LCP_OWN);
+    return_if_error (rc, "setup_nv for INDEX_LCP_OWN");
+    rc = setup_nv (sys_ctx, INDEX_LCP_SUP);
+    return_if_error (rc, "setup_nv for INDEX_LCP_SUP");
```

Comment:
If the second setup_nv() fails, or in case of any other failure along the way, we wont clean the INDEX_LCP_OWN NV index. Instead of return_if_error let's use "goto error" and call teardown_nv() for both indexes always.

Reply:
Sounds like the requirement here is to clean up the NV regions created in the case of a test failure? I didn't do exactly what you suggest but I think the outcome is the same.

Reply:
That looks good. Thanks @flihp 
Can we do the same in sapi-nv-policy-locality.int.c (next commit)

Reply:
Just pushed changes to the policy-locality test @tstruk, rebased as well.